tests/testthat/test-interface.R

```diff
@@ -57,3 +56,30 @@ test_that("cache hits don't compile", {
   mockery::expect_called(mock_compile_mode, 1)
   expect_identical(gen3, gen)
 })
+
+test_that("Can update time", {
+  path <- mode_file("examples/logistic.cpp")
+  gen <- mode(path, quiet = TRUE)
+  pars <- list(r1 = 0.1, r2 = 0.2, K1 = 100, K2 = 100)
+  n_particles <- 10
+  initial_time <- 1
+  mod <- gen$new(pars, initial_time, n_particles)
+  expect_equal(mod$time(), initial_time)
+  res <- mod$solve(5)
+  expect_equal(mod$time(), 5)
+  mod$update_state(initial_time)
+  expect_equal(mod$time(), initial_time)
+  res2 <- mod$solve(5)
+  expect_equal(res, res2)
```

Comment:
Might be worth expect_identical here to check that the initial step size would have been reset rtoo

Reply:
ah, I failed to appreciate that `expect_equal` meant something different, yep!